### PR TITLE
[nrf fromlist] soc: arm: nrf5340: Add option to allow the app core to run at 128MHz

### PR DESCRIPTION
**Pull request by jostn**
Add KConfig option to allow the app core to run at 128MHz

Signed-off-by: Johan Stridkvist <johan.stridkvist@nordicsemi.no>

### CODE REVIEW

File: boards/arm/nrf5340dk_nrf5340/Kconfig [hunk 1]
```diff
@@ -56,6 +56,11 @@ config BOARD_ENABLE_DCDC_HV
 	select SOC_DCDC_NRF53X_HV
 	default y
 
+config BOARD_ENABLE_128MHZ_APP
+	bool "Enable 128MHz speed in application core"
+	select SOC_CLOCK_128MHZ_NRF53X_APP
+	default n
```
Comment (ioannisg) [inline boards/arm/nrf5340dk_nrf5340/Kconfig:62]: no need for default n, this is default anways

File: soc/arm/nordic_nrf/nrf53/Kconfig.soc [hunk 2]
```diff
@@ -130,6 +130,12 @@ config SOC_DCDC_NRF53X_HV
 	help
 	  Enable nRF53 series System on Chip High Voltage DC/DC converter.
 
+config SOC_CLOCK_128MHZ_NRF53X_APP
+	bool
+	help
+	  Run nRF53 series app core at 128MHz.
+	default n
```
Comment (ioannisg) [inline soc/arm/nordic_nrf/nrf53/Kconfig.soc:137]: ditto

**Review by carlescufi — CHANGES_REQUESTED** [review]
This should be `fromtree` or `fromlist`

**Comment by mbolivar-nordic** [comment]
Please rebase if still needed downstream, as this is not yet merged upstream

**Pull request closed without merging**